module/grid/src/Column/Renderer/LinkColumnRenderer.php

```diff
@@ -82,7 +85,10 @@ private function isVisible(array $link, array $row): bool
     {
         if (array_key_exists('privilege', $link)) {
             $privilege = $link['privilege'];
-            if (!$this->checker->isGranted($privilege)) {
+            if (!array_key_exists($privilege, $this->localCache)) {
+                $this->localCache[$privilege] = $this->checker->isGranted($privilege);
+            }
+            if (!$this->localCache[$privilege]) {
                 return false;
```

Comment:
Since you are donig optimisation worth removing line 100 and changing 104 to `return false;`